lib/msprime.h

```diff
@@ -123,13 +123,31 @@ typedef struct {
     double c; // constant
 } dirac_coalescent_t;
 
+/* Forward declaration */
+struct _msp_t;
+
 typedef struct {
-    uint32_t locus;
+    double start_frequency;
+    double end_frequency;
+    double alpha;
+    double dt;
+} genic_selection_trajectory_t;
+
+typedef struct _single_sweep_t {
+    /* Deprecated - remove this if we go with the generic inferface here. */
     struct {
         double *allele_frequency;
         double *time;
         size_t num_steps;
     } trajectory;
+
+    uint32_t locus;
+    union {
+        /* Other selection trajectory simulators would go here */
+        genic_selection_trajectory_t genic_selection_trajectory;
+    } trajectory_params;
+    int (*generate_trajectory)(struct _single_sweep_t *self, struct _msp_t *simulator,
```

Comment:
This is the declaration of the function pointer that for the single_sweep_t simulation model. I'm imagining this is set in the ``set_single_sweep_model`` code path somehow.